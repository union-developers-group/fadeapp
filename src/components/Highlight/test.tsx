import { render, screen } from '@testing-library/react'

import { Highlight, HighlightProps } from '.'

const { getByRole } = screen

const highlight: HighlightProps = {
  highlightItems: [
    {
      id: '0',
      title: 'Title',
      text: 'Text',
      image: 'http://cdn.com/photo-name.jpg',
    },
    {
      id: '1',
      title: 'Title2',
      text: 'Text2',
      image: 'http://cdn.com/photo2-name.jpg',
    },
    {
      id: '2',
      title: 'Title3',
      text: 'Text3',
      image: 'http://cdn.com/photo3-name.jpg',
    },
  ],
}

describe('<Highlight />', () => {
  it('should render by default', () => {
    const { container } = render(<Highlight {...highlight} />)

    const image = getByRole('img', { name: 'Title' })
    const highlightContainer = container.firstChild

    expect(image).toHaveAttribute('alt', 'Title')
    expect(image).toHaveAttribute('src', 'http://cdn.com/photo-name.jpg')
    expect(highlightContainer).not.toHaveClass(
      'after:content-[""] after:max-w-[55%] after:absolute after:bg-primary after:bottom-0 after:h-1 after:w-3/5 lg:after:w-4/5 after:last:hidden',
    )
    expect(highlight.highlightItems.length).toBe(3)
  })

  it('should render with line', () => {
    const { container } = render(<Highlight {...highlight} withLine />)

    const highlightContainer = container.firstChild

    expect(highlightContainer).toHaveClass(
      'after:content-[""] after:max-w-[55%] after:absolute after:bg-primary after:bottom-0 after:h-1 after:w-3/5 lg:after:w-4/5 after:last:hidden',
    )
  })

  it('should render with inverterOrder actived', () => {
    const { container } = render(<Highlight {...highlight} inverterOrder />)

    const highlightContainer = container.firstChild

    expect(highlightContainer).toHaveClass('lg:odd:flex-row-reverse')
  })

  it('should render with line and inverterOrder actived', () => {
    const { container } = render(
      <Highlight {...highlight} withLine inverterOrder />,
    )

    const highlightContainer = container.firstChild

    expect(highlightContainer).toHaveClass(
      'after:odd:left-0 after:even:right-0',
    )
  })

  it('Should match snapshot', () => {
    const { container } = render(<Highlight {...highlight} />)

    expect(container).toMatchSnapshot()
  })
})
