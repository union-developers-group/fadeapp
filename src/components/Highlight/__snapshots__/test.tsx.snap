// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> Should match snapshot 1`] = `
<div>
  <div
    class="
relative 
flex 
items-center 
justify-center 
w-full 
max-w-screen 
gap-20 
lg:gap-14 
xl:gap-32 
px-4 
sm:px-8 
py-[3.75rem] 
mx-auto 
flex-col 
lg:flex-row
lg:even:flex-row-reverse 
         "
  >
    <div
      class="
text-white 
text-center 
lg:text-left 
max-w-md 
flex-1
"
    >
      <h3
        class="
text-4xl 
sm:text-6xl 
lg:text-7xl 
mb-4 
sm:mb-6 
font-bold
"
      >
        Title
      </h3>
      <p
        class="
text-base 
sm:text-xl 
lg:text-2xl
"
      >
        Text
      </p>
    </div>
    <div
      class="
box-border 
flex 
justify-center 
items-center 
max-w-[80%]
lg:max-w-[50%] 
xl:max-w-full
"
    >
      <img
        alt="Title"
        class="
w-full 
h-full
"
        src="http://cdn.com/photo-name.jpg"
      />
    </div>
  </div>
  <div
    class="
relative 
flex 
items-center 
justify-center 
w-full 
max-w-screen 
gap-20 
lg:gap-14 
xl:gap-32 
px-4 
sm:px-8 
py-[3.75rem] 
mx-auto 
flex-col 
lg:flex-row
lg:even:flex-row-reverse 
         "
  >
    <div
      class="
text-white 
text-center 
lg:text-left 
max-w-md 
flex-1
"
    >
      <h3
        class="
text-4xl 
sm:text-6xl 
lg:text-7xl 
mb-4 
sm:mb-6 
font-bold
"
      >
        Title2
      </h3>
      <p
        class="
text-base 
sm:text-xl 
lg:text-2xl
"
      >
        Text2
      </p>
    </div>
    <div
      class="
box-border 
flex 
justify-center 
items-center 
max-w-[80%]
lg:max-w-[50%] 
xl:max-w-full
"
    >
      <img
        alt="Title2"
        class="
w-full 
h-full
"
        src="http://cdn.com/photo2-name.jpg"
      />
    </div>
  </div>
  <div
    class="
relative 
flex 
items-center 
justify-center 
w-full 
max-w-screen 
gap-20 
lg:gap-14 
xl:gap-32 
px-4 
sm:px-8 
py-[3.75rem] 
mx-auto 
flex-col 
lg:flex-row
lg:even:flex-row-reverse 
         "
  >
    <div
      class="
text-white 
text-center 
lg:text-left 
max-w-md 
flex-1
"
    >
      <h3
        class="
text-4xl 
sm:text-6xl 
lg:text-7xl 
mb-4 
sm:mb-6 
font-bold
"
      >
        Title3
      </h3>
      <p
        class="
text-base 
sm:text-xl 
lg:text-2xl
"
      >
        Text3
      </p>
    </div>
    <div
      class="
box-border 
flex 
justify-center 
items-center 
max-w-[80%]
lg:max-w-[50%] 
xl:max-w-full
"
    >
      <img
        alt="Title3"
        class="
w-full 
h-full
"
        src="http://cdn.com/photo3-name.jpg"
      />
    </div>
  </div>
</div>
`;
